### PR TITLE
Feature/jjdd8 hfv 33

### PR DESCRIPTION
**Pull request by dagiel**
Classes done for Employee and Team:
Add/Edit/Delete Servlet->requestMapper->Request.->Service->entityMapper->dao
Missing:
1) add/edit/delete methods in dao classes only initialized - to be covered by logic
2) add-team.html and add-employee.html to be finished and transformed to Freemarker templates
3) templates to be input into servlets.

### CODE REVIEW

**Comment by isacademyci** [comment]
Build triggered for merge commit.

**Comment by isacademyci** [comment]
Build started for merge commit.

**Comment by isacademyci** [comment]
Build finished.

**Comment by dagiel** [comment]
one more comment: feature#33 merged with latest version of #32 and # 34 before PR

**Comment by isacademyci** [comment]
Build triggered for merge commit.

**Comment by isacademyci** [comment]
Build started for merge commit.

**Comment by isacademyci** [comment]
Build finished.

**Comment by isacademyci** [comment]
Build triggered for merge commit.

**Comment by isacademyci** [comment]
Build started for merge commit.

**Comment by isacademyci** [comment]
Build finished.

File: web/src/main/java/com/infoshare/academy/highfive/mapper/entity/EmployeeMapper.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import javax.enterprise.context.RequestScoped;
+
+@RequestScoped
+public class EmployeeMapper {
+
+    public Employee mapRequestToEntity(EmployeeRequest request){
+
+       Employee employee = new Employee();
+
+        employee.setId(request.getId());
```
Comment (rafalmisiak) [inline web/src/main/java/com/infoshare/academy/highfive/mapper/entity/EmployeeMapper.java]: formatowanie

File: web/src/main/java/com/infoshare/academy/highfive/mapper/entity/TeamMapper.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+import javax.enterprise.context.RequestScoped;
+
+@RequestScoped
+public class TeamMapper {
+
+    public Team mapRequestToEntity(TeamRequest request){
+
+   Team team = new Team();
+        team.setId(request.getId());
```
Comment (rafalmisiak) [inline web/src/main/java/com/infoshare/academy/highfive/mapper/entity/TeamMapper.java]: formatowanie

File: web/src/main/java/com/infoshare/academy/highfive/servlet/teamServlets/EditTeamServlet.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import javax.ejb.EJB;
+import javax.inject.Inject;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+@WebServlet("/manager/add-team/")
```
Comment (rafalmisiak) [inline web/src/main/java/com/infoshare/academy/highfive/servlet/teamServlets/EditTeamServlet.java]: ten kod się nie uruchamia, powtórzyłeś wiele razy te same konteksty webowe, również w innych servletach

File: web/src/main/java/com/infoshare/academy/highfive/dao/TeamDao.java [hunk 4]
```diff
@@ -0,0 +1,15 @@
+package com.infoshare.academy.highfive.dao;
+
+import com.infoshare.academy.highfive.domain.Team;
+
+import javax.ejb.Stateless;
+
+@Stateless
+public class TeamDao {
+
+    public void addTeam(Team team){}
```
Comment (rafalmisiak) [inline web/src/main/java/com/infoshare/academy/highfive/dao/TeamDao.java]: Jeśli mamy już TeamDao to metody nie muszą mieć informacji że modyfikują Team. Wszystko już jest jasne - dt każdego DAO :)

**Comment by isacademyci** [comment]
Build triggered for merge commit.

**Comment by isacademyci** [comment]
Build started for merge commit.

**Comment by isacademyci** [comment]
Build finished.

File: web/src/main/java/com/infoshare/academy/highfive/mapper/entity/EmployeeMapper.java (hunk 1 above)
Comment (dagiel) [reply web/src/main/java/com/infoshare/academy/highfive/mapper/entity/EmployeeMapper.java]: poprawione. dzięki

File: web/src/main/java/com/infoshare/academy/highfive/mapper/entity/TeamMapper.java (hunk 2 above)
Comment (dagiel) [reply web/src/main/java/com/infoshare/academy/highfive/mapper/entity/TeamMapper.java]: poprawione. dzięki

File: web/src/main/java/com/infoshare/academy/highfive/servlet/teamServlets/EditTeamServlet.java (hunk 3 above)
Comment (dagiel) [reply web/src/main/java/com/infoshare/academy/highfive/servlet/teamServlets/EditTeamServlet.java]: poprawione. dzięki

Comment (dagiel) [reply web/src/main/java/com/infoshare/academy/highfive/servlet/teamServlets/EditTeamServlet.java]: poprawione. dzięki

File: web/src/main/java/com/infoshare/academy/highfive/dao/TeamDao.java (hunk 4 above)
Comment (dagiel) [reply web/src/main/java/com/infoshare/academy/highfive/dao/TeamDao.java]: poprawione. dzięki

**Comment by dagiel** [comment]
All Rafal's comments included

**Comment by isacademyci** [comment]
Build triggered for merge commit.

**Comment by isacademyci** [comment]
Build started for merge commit.

**Comment by isacademyci** [comment]
Build finished.

File: web/src/main/java/com/infoshare/academy/highfive/mapper/request/TeamRequestMapper.java [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+import javax.servlet.http.HttpServletRequest;
+
+@RequestScoped
+public class TeamRequestMapper {
+
+  public TeamRequest mapParamsToRequest(HttpServletRequest request) {
+
+    TeamRequest teamRequest = new TeamRequest();
+
+    teamRequest.setId((Long) request.getAttribute("team_id"));
```
Comment (rafalmisiak) [inline web/src/main/java/com/infoshare/academy/highfive/mapper/request/TeamRequestMapper.java]: czy możemy Object.valueOf i String.valueOf ? i we wszystkich innych miejscach gdzie pobierasz Atrybut

File: web/src/main/java/com/infoshare/academy/highfive/servlet/employeeServlets/AddEmployeeServlet.java [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    private EmployeeService employeeService;
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+
+        try {
+            EmployeeRequest employeeRequest = requestMapper.mapParamsToRequest(req);
+            employeeService.addNewEmployee(employeeRequest);
+        } catch (ParseException e) {
+            e.printStackTrace();
```
Comment (rafalmisiak) [inline web/src/main/java/com/infoshare/academy/highfive/servlet/employeeServlets/AddEmployeeServlet.java]: logujemy nie wypisujemy

File: web/src/main/java/com/infoshare/academy/highfive/servlet/employeeServlets/DeleteEmployeeServlet.java [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+    private EmployeeService employeeService;
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+
+        try {
+            EmployeeRequest employeeRequest = requestMapper.mapParamsToRequest(req);
+            employeeService.deleteEmployee(employeeRequest);
+        } catch (ParseException e) {
+            e.printStackTrace();
```
Comment (rafalmisiak) [inline web/src/main/java/com/infoshare/academy/highfive/servlet/employeeServlets/DeleteEmployeeServlet.java]: jw

File: web/src/main/java/com/infoshare/academy/highfive/servlet/employeeServlets/EditEmployeeServlet.java [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+    private EmployeeService employeeService;
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+
+        try {
+            EmployeeRequest employeeRequest = requestMapper.mapParamsToRequest(req);
+            employeeService.editEmployee(employeeRequest);
+        } catch (ParseException e) {
+            e.printStackTrace();
```
Comment (rafalmisiak) [inline web/src/main/java/com/infoshare/academy/highfive/servlet/employeeServlets/EditEmployeeServlet.java]: jw

File: web/src/main/java/com/infoshare/academy/highfive/servlet/teamServlets/EditTeamServlet.java [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+    Logger logger = LoggerFactory.getLogger(getClass().getName());
+
+    @Inject
+    private TeamRequestMapper requestMapper;
+
+    @EJB
+    private TeamService teamService;
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
```
Comment (rafalmisiak) [inline web/src/main/java/com/infoshare/academy/highfive/servlet/teamServlets/EditTeamServlet.java]: Dlaczego edit-team robimy postem ? wszystkie edycje PUTem

**Comment by isacademyci** [comment]
Build triggered for merge commit.

**Comment by isacademyci** [comment]
Build started for merge commit.

**Comment by isacademyci** [comment]
Build finished.

**Comment by isacademyci** [comment]
Build triggered for merge commit.

**Comment by isacademyci** [comment]
Build started for merge commit.

**Comment by isacademyci** [comment]
Build finished.

**Comment by isacademyci** [comment]
Build triggered for merge commit.

**Comment by isacademyci** [comment]
Build started for merge commit.

**Comment by isacademyci** [comment]
Build finished.

**Comment by isacademyci** [comment]
Build triggered for merge commit.

**Comment by isacademyci** [comment]
Build started for merge commit.

**Comment by isacademyci** [comment]
Build finished.

**Comment by isacademyci** [comment]
Build triggered for merge commit.

**Comment by isacademyci** [comment]
Build started for merge commit.

**Comment by isacademyci** [comment]
Build finished.

**Comment by isacademyci** [comment]
Build triggered for merge commit.

**Comment by isacademyci** [comment]
Build started for merge commit.

**Comment by isacademyci** [comment]
Build finished.

**Pull request closed without merging**